### DISCUSSION TITLE
Link Prediction using Node Features - Help with Tensor Representation

### DISCUSSION THREAD

**Discussion by mahadafzal** [pyg-team/pytorch_geometric#2340] [Q&A]
I am trying to tackle a Link Prediction Task and following the example given in the repository on the CORA dataset. I have represented each sample (a user comment) in the dataset as a Graph, where each node is a sub-sentence, with its features being the text of the sub-sentence, and the edges representing a link between the sub-sentences. The CORA example uses BoW, and I have followed as such. I am trying to predict the edges given the node features. 

However, I am running into Tensor Representation errors (Float vs Long). I found that a similar issue was posted and added a AtomEncoder, but to no avail. Could someone guide me in the right direction? 

Any suggestions, specific or general, would be much appreciated.

**Comment by rusty1s** [comment]
You need to make sure that node features `x` are of type `float`, and edge connectivity `edge_index` is of type `long`.

**Reply by mahadafzal** [reply]
Thank you! After ensuring the types of x and edge_index were float and long respectively, I found that my issue was empty edge_index tensors, a solution to which was suggested in issue [#2064](https://github.com/rusty1s/pytorch_geometric/issues/2064)
